### PR TITLE
Config path to Conan exe, confirm dialog before wiping the Cache, other improvements

### LINKED ISSUE #25: Conan plugin's actions throw exceptions when conan executable isn't configured
- install Conan plugin
- click Install button

```stacktrace
Details: [com.intellij.execution.configurations.GeneralCommandLine.createProcess(GeneralCommandLine.java:416), com.intellij.execution.process.OSProcessHandler.startProcess(OSProcessHandler.java:51), com.intellij.execution.process.OSProcessHandler.<init>(OSProcessHandler.java:44), conan.commands.task.SyncConanTask.run(SyncConanTask.java:44), com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:731), com.intellij.openapi.progress.impl.CoreProgressManager$5.run(CoreProgressManager.java:442), com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164), com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:586), com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532), com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:86), com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151), com.intellij.openapi.application.impl.ApplicationImpl.lambda$null$10(ApplicationImpl.java:594), com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:311), java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511), java.util.concurrent.FutureTask.run(FutureTask.java:266), java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149), java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624), java.lang.Thread.run(Thread.java:748)]

java.lang.Throwable: Cannot run program "conan" (in directory "C:\Users\Maria.Baburina\ClionProjects\untitled34"): CreateProcess error=2, The system cannot find the file specified
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:162)
	at conan.utils.Utils.log(Utils.java:38)
	at conan.commands.task.SyncConanTask.run(SyncConanTask.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:731)
	at com.intellij.openapi.progress.impl.CoreProgressManager$5.run(CoreProgressManager.java:442)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:586)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:86)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$null$10(ApplicationImpl.java:594)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:311)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

- click 'Clean console' action

```stacktrace
java.lang.NullPointerException
	at conan.ui.ConanToolWindow.cleanConsole(ConanToolWindow.java:160)
	at conan.actions.CleanConsole.actionPerformed(CleanConsole.java:24)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:260)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:277)
	at com.intellij.openapi.actionSystem.impl.ActionButton.actionPerformed(ActionButton.java:183)
	at com.intellij.openapi.actionSystem.impl.ActionButton.performAction(ActionButton.java:147)
	at com.intellij.openapi.actionSystem.impl.ActionButton.processMouseEvent(ActionButton.java:415)
	at java.awt.Component.processEvent(Component.java:6315)
	at java.awt.Container.processEvent(Container.java:2239)
	at java.awt.Component.dispatchEventImpl(Component.java:4899)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Component.dispatchEvent(Component.java:4721)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
	at java.awt.Container.dispatchEventImpl(Container.java:2283)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4721)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:766)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
	at java.awt.EventQueue$4.run(EventQueue.java:739)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:736)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:747)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:692)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

### PR DESCRIPTION
**Pull request by lasote**
- Allowed to configure the Conan executable path.
- Fixed exceptions raised and other unexpected behaviors when no conan executable is found.
- Added confirmation dialog before wiping the local cache.

Close #25

### CODE REVIEW

**Comment by lasote** [comment]
@eyalbe4 @yahavi we want to merge this to solve the reported issues with the Conan path ASAP. Would be nice if you could take a look.

File: src/main/java/conan/actions/ActionUtils.java [hunk 1]
```diff
@@ -30,7 +37,11 @@
      * @param update    true if it's update and install action.
      */
     static void runInstall(Project project, Component component, boolean update) {
-        new Version(project).run();
+
+        if(!isConanInstalled(project)){
```
Comment (eyalbe4) [inline src/main/java/conan/actions/ActionUtils.java]: White-space should be added after the "if" to be compatible with the existing code conventions.
Let's also remove the first new-line in the method.

File: src/main/java/conan/actions/ActionUtils.java [hunk 2]
```diff
@@ -51,6 +62,11 @@ static void runInstall(Project project, Component component, boolean update) {
      * @param update    true if it's update and install action.
      */
     private static void matchProfilesAndInstall(Project project, Component component, boolean update) {
+
+        if(!isConanInstalled(project)){
```
Comment (eyalbe4) [inline src/main/java/conan/actions/ActionUtils.java]: White-space should be added after the "if" to be compatible with the existing code conventions.
Let's also remove the first new-line in the method.

File: src/main/java/conan/actions/CleanCacheAction.java [hunk 3]
```diff
@@ -13,8 +19,17 @@
...
-        if (getEventProject(anActionEvent) != null) {
-            new conan.commands.CleanCache(anActionEvent.getProject()).run();
+        Project project = getEventProject(anActionEvent);
+        if (project != null) {
+            if (!ActionUtils.isConanInstalled(project)) {
+                return;
+            }
+            String message = "This will remove the Conan local cache, Are you sure?";
+            boolean result = new ConanConfirmDialog("Removing Conan Cache", message).showAndGet();
+            if(result) {
```
Comment (eyalbe4) [inline src/main/java/conan/actions/CleanCacheAction.java]: White-space, similar to the the other comments.

File: src/main/java/conan/actions/MatchProfilesAction.java [hunk 4]
```diff
@@ -19,6 +19,9 @@ public void actionPerformed(AnActionEvent anActionEvent) {
         if (project == null) {
             return;
         }
+        if(!ActionUtils.isConanInstalled(project)){
```
Comment (eyalbe4) [inline src/main/java/conan/actions/MatchProfilesAction.java]: White-space, similar to the the other comments.

File: src/test/java/conan/commands/ListProfilesTest.java [hunk 5]
```diff
@@ -20,8 +20,8 @@
 
     @Test
     public void testListProfiles() {
-        verifyProfiles(Sets.newHashSet());
+        //verifyProfiles(Sets.newHashSet());
```
Comment (eyalbe4) [inline src/test/java/conan/commands/ListProfilesTest.java]: Why were these tests removed?

File: src/main/java/conan/ui/configuration/ConanConfig.java [hunk 6]
```diff
@@ -41,11 +49,16 @@
     private JBTextField installArgs;
     private JLabel installArgsLabel;
 
+    private JLabel conanPathLabel;
+    private TextFieldWithBrowseButton conanPath;
+
```
Comment (yahavi) [inline src/main/java/conan/ui/configuration/ConanConfig.java]:
```suggestion
```

File: src/main/java/conan/ui/configuration/ConanConfig.java [hunk 7]
```diff
@@ -71,6 +84,21 @@ public JComponent createComponent() {
             runConfigInstall(url);
         });
         installArgs.getEmptyText().setText("Arguments other than '--if', '--pr' and '--update'");
+        String envExePath = System.getenv("CONAN_EXE_PATH");
+        if(envExePath != null) {
```
Comment (yahavi) [inline src/main/java/conan/ui/configuration/ConanConfig.java]:
```suggestion
        if (envExePath != null) {
```

File: src/main/java/conan/actions/ActionUtils.java [hunk 8]
```diff
@@ -5,6 +5,7 @@
 import com.intellij.openapi.fileEditor.FileDocumentManager;
 import com.intellij.openapi.project.Project;
 import conan.commands.Install;
+import conan.commands.IsInstalledCommand;
 import conan.commands.Version;
```
Comment (yahavi) [inline src/main/java/conan/actions/ActionUtils.java]: Unused import

File: src/main/java/conan/actions/ActionUtils.java [hunk 9]
```diff
@@ -30,7 +37,11 @@
      * @param update    true if it's update and install action.
      */
     static void runInstall(Project project, Component component, boolean update) {
-        new Version(project).run();
+
```
Comment (yahavi) [inline src/main/java/conan/actions/ActionUtils.java]:
```suggestion
```

File: src/main/java/conan/actions/ActionUtils.java [hunk 10]
```diff
@@ -51,6 +62,11 @@ static void runInstall(Project project, Component component, boolean update) {
      * @param update    true if it's update and install action.
      */
     private static void matchProfilesAndInstall(Project project, Component component, boolean update) {
+
```
Comment (yahavi) [inline src/main/java/conan/actions/ActionUtils.java]:
```suggestion
```

File: src/main/java/conan/actions/CleanCacheAction.java [hunk 11]
```diff
@@ -3,6 +3,12 @@
 import com.intellij.openapi.actionSystem.AnAction;
 import com.intellij.openapi.actionSystem.AnActionEvent;
 import com.intellij.openapi.project.DumbAware;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
```
Comment (yahavi) [inline src/main/java/conan/actions/CleanCacheAction.java]: Please optimize imports

File: src/test/java/conan/commands/ConfigInstallTest.java [hunk 12]
```diff
@@ -9,10 +9,11 @@
 @Test
 public class ConfigInstallTest extends ConanCommandTestBase {
 
+
```
Comment (yahavi) [inline src/test/java/conan/commands/ConfigInstallTest.java]:
```suggestion
```

File: src/test/java/conan/commands/ListProfilesTest.java (hunk 5 above)
Comment (lasote) [reply src/test/java/conan/commands/ListProfilesTest.java]: Thanks, forgot to uncomment after the refactor

**Pull request merged**